src/mito2/src/worker/handle_open.rs

```diff
@@ -36,6 +38,18 @@ impl<S: LogStore> RegionWorkerLoop<S> {
             return Ok(Output::AffectedRows(0));
         }
 
+        // Check if this region is pending drop. And clean the entire dir if so.
+        if !self.dropping_regions.is_region_exists(region_id)
+            && self
+                .object_store
+                .is_exist(&join_path(&request.region_dir, DROPPING_MARKER_FILE))
+                .await
+                .context(OpenDalSnafu)?
+        {
+            let _ = self.object_store.delete(&request.region_dir).await;
```

Comment:
issue:
1. `delete()` might only deletes one object, not the whole directory
2. If the marker file is deleted before deleting other files, then we won't be able to remove the directory next time.